language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/inlayhint/InlayHintProvider.java

```diff
@@ -128,12 +146,58 @@ private static Pair<List<ParameterSymbol>, Optional<ParameterSymbol>> getParamet
             // Since the method is a lang-lib function, skip the first parameter
             return Pair.of(libFunction.get().params().get().stream().skip(1).collect(Collectors.toList()),
                     libFunction.get().restParam());
+        } else if (invokableNode.kind() == SyntaxKind.CLIENT_RESOURCE_ACCESS_ACTION) {
+            ClientResourceAccessActionNode resourceAccessActionNode = (ClientResourceAccessActionNode) invokableNode;
+            Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(resourceAccessActionNode);
+            FunctionTypeSymbol typeSymbol = ((ResourceMethodSymbol) symbol.get()).typeDescriptor();
+            return symbol.map(methodSymbol -> Pair.of(typeSymbol.params().orElse(Collections.emptyList()),
+                            typeSymbol.restParam())).orElse(Pair.of(Collections.emptyList(), Optional.empty()));
+        } else if (invokableNode.kind() == SyntaxKind.REMOTE_METHOD_CALL_ACTION) {
+            RemoteMethodCallActionNode methodCallActionNode = (RemoteMethodCallActionNode) invokableNode;
+            Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(methodCallActionNode);
+            List<ParameterSymbol> parameterSymbols = ((MethodSymbol) symbol.get()).typeDescriptor()
+                    .params().get();
+            return Pair.of(parameterSymbols, Optional.empty());
+        } else if (invokableNode.kind() == SyntaxKind.IMPLICIT_NEW_EXPRESSION) {
+            return getParameterSymbolsForNewExpressions(context, invokableNode);
+        } else if (invokableNode.kind() == SyntaxKind.EXPLICIT_NEW_EXPRESSION) {
+            return getParameterSymbolsForNewExpressions(context, invokableNode);
```

Comment:
Since we are again checking for implicit/explicit expression within `getParameterSymbolsForNewExpressions` shall we create 2 separate methods for this?

Reply:
+1